### PR TITLE
[AM][OPS-9610] add left over income to journey

### PR DESCRIPTION
**Pull request by almorcrette**
Add total left over income to journey via affordability controller to be used as input to payment plan calculations

### CODE REVIEW

File: app/ssttpaffordability/AffordabilityController.scala [hunk 1]
```diff
@@ -88,20 +88,32 @@ class AffordabilityController @Inject() (
...
       }
     }
   }
 
+  def saveTotalLeftOverIncome: Action[AnyContent] = as.authorisedSaUser.async { implicit request =>
+    JourneyLogger.info(s"AffordabilityController.getHowMuchYouCouldAfford: $request")
+    journeyService.authorizedForSsttp { journey =>
+      val spending = journey.maybeSpending.fold(Seq.empty[Expenses])(_.expenses)
+      val income = journey.maybeIncome.fold(Seq.empty[IncomeBudgetLine])(_.budgetLines)
+      val totalLeftOverIncome = income.map(_.amount).sum - spending.map(_.amount).sum
```
Comment (andy1138) [inline app/ssttpaffordability/AffordabilityController.scala]: Can this now be  `val totalLeftOverIncome = journey.totalLeftOverIncome` ? (keeping the calculation in one place )

File: app/ssttpaffordability/AffordabilityController.scala [hunk 2]
```diff
@@ -88,20 +88,32 @@ class AffordabilityController @Inject() (
...
+  }
+
   def getHowMuchYouCouldAfford: Action[AnyContent] = as.authorisedSaUser.async { implicit request =>
     JourneyLogger.info(s"AffordabilityController.getHowMuchYouCouldAfford: $request")
     journeyService.authorizedForSsttp { journey =>
       val spending = journey.maybeSpending.fold(Seq.empty[Expenses])(_.expenses)
       val income = journey.maybeIncome.fold(Seq.empty[IncomeBudgetLine])(_.budgetLines)
-      val total = income.map(_.amount).sum - spending.map(_.amount).sum
-      Future.successful(Ok(views.how_much_you_could_afford(income, spending, total)))
+      val totalLeftOverIncome = journey.totalLeftOverIncome
```
Comment (andy1138) [inline app/ssttpaffordability/AffordabilityController.scala]: ( in all paces that have `totalLeftOverIncome` )

I first read this as 'left over from what?' Is it from 'remainingIncome'? or `remainingIncomeAfterSpending` (that's way too long :-) )

Comment (almorcrette) [reply app/ssttpaffordability/AffordabilityController.scala]: Yeah, happy to change the name. Yours is better

File: app/ssttpaffordability/AffordabilityController.scala (hunk 1 above)
Comment (almorcrette) [reply app/ssttpaffordability/AffordabilityController.scala]: At that point it's not saved in the journey. This method save it to the journey before redirecting to displaying the next page...

Am discussing keeping the calculation in one place on Slack with Kostas. I did try that but got a type return mismatch. What am I doing wrong?

I got a type return mismatch.
Found `Unit => Future[Result]`
Required `Unit => S_`

```scala
  def getHowMuchYouCouldAfford: Action[AnyContent] = as.authorisedSaUser.async { implicit request =>
    JourneyLogger.info(s"AffordabilityController.getHowMuchYouCouldAfford: $request")
    journeyService.authorizedForSsttp { journey =>
      val spending = journey.maybeSpending.fold(Seq.empty[Expenses])(_.expenses)
      val income = journey.maybeIncome.fold(Seq.empty[IncomeBudgetLine])(_.budgetLines)
      val totalLeftOverIncome = income.map(_.amount).sum - spending.map(_.amount).sum
      storeLeftOverIncomeToJourney(totalLeftOverIncome, journey).map { _ =>
      Future.successful(Ok(views.how_much_you_could_afford(income, spending, totalLeftOverIncome)))
    }
  }
```

Comment (almorcrette) [reply app/ssttpaffordability/AffordabilityController.scala]: Kostas has helped me with a solution

**Pull request closed without merging**